invenio_rdm_records/records/systemfields/deletion_status.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""System field for managing the record deletion status."""
+
+import enum
+
+from invenio_records.systemfields import SystemField
+
+
+class RecordDeletionStatusEnum(enum.Enum):
+    """Enumeration of a record's possible deletion states."""
+
+    PUBLISHED = "P"
+    DELETED = "D"
+    MARKED = "X"
+
+
+class RecordDeletionStatus:
+    """The deletion status of the record."""
+
+    def __init__(self, status):
+        """Constructor."""
+        self.status = status
+
+    @property
+    def status(self):
+        """Get the deletion status."""
+        return self._status.value
+
+    @status.setter
+    def status(self, value):
+        """Set the deletion status."""
+        if value is None:
+            self._status = RecordDeletionStatusEnum.PUBLISHED
+
+        elif isinstance(value, str):
+            self._status = RecordDeletionStatusEnum(value)
+
+        elif isinstance(value, RecordDeletionStatusEnum):
+            self._status = value
+
+        else:
+            raise ValueError(f"Invalid value for record deletion status: {value}")
+
+    @property
+    def is_deleted(self):
+        """Check if the record is deleted."""
+        return self._status != RecordDeletionStatusEnum.PUBLISHED
+
+    def __repr__(self):
+        """Return repr(self)."""
+        return repr(self._status)
+
+    def __str__(self):
+        """Return str(self)."""
+        return self.status
+
+
+class RecordDeletionStatusField(SystemField):
+    """System field for accessing a record's deletion status."""
+
+    def __init__(self, record_cls=None, key=None):
+        """Constructor."""
+        super().__init__(key=key)
+        self.record_cls = record_cls
```

Comment:
```suggestion
    def __init__(self, key=None):
        """Constructor."""
        super().__init__(key=key)
```

minor: is `self.record_cls` used anywhere? I don't see it being passed in the API class where the systemfield is defined/used.

If that is the case, then maybe the constructor is not needed at all.